src/GraphQL/GraphQLExtensions.cs

```diff
@@ -470,5 +476,40 @@ public static IValue ToAST(this IGraphType type, object value)
 
             throw new ArgumentOutOfRangeException(nameof(type), $"Must provide Input Type, cannot use {type.GetType().Name} '{type}'");
         }
+
+        /// <summary>
+        /// Returns a string representation of the specified node.
+        /// </summary>
+        public static string Print(this ASTNode node)
+        {
+            var context = new PrintContext();
+            _sdlWriter.Visit(node, context).GetAwaiter().GetResult(); // actually is sync
+            return context.Writer.ToString()!;
+        }
+
+        private static readonly SDLWriterEx _sdlWriter = new();
+
+        private class SDLWriterEx : SDLWriter<PrintContext>
+        {
+            public override ValueTask Visit(ASTNode? node, PrintContext context)
+            {
+                return node switch //TODO: deal with AnyValue
+                {
+                    AnyValue _ => throw new InvalidOperationException("TODO: may be implemented? see anynode_throws test"),
```

Comment:
I don't like this code at first glance.  Why doesn't AnyValue handle it's own printing, and can throw an exception or whatever from inside it?  Should consider redesigning printing code in parser project so printing code for each node is contained within each class.  Another possibility is that an exception should be thrown within the printer if it encounters a node it doesn't recognize.  It would keep printing code together, and (if members are virtual) allow for overriding the printer to change behavior.  In any case I'm not sure why we need a visitor to examine the document before printing.

Reply:
> I don't like this code at first glance

Either I.

> Why doesn't AnyValue handle it's own printing, and can throw an exception or whatever from inside it? 

I don't know. I just added line of code that equals to previous implementation  - throw exception.

>  Should consider redesigning printing code in parser project so printing code for each node is contained within each class.

In fact we **can** get rid of `AstKind` enum in parser 😄 . It should not be so difficult.

Reply:
I removed that code.